lightning/backend/backend.go

```diff
@@ -104,7 +106,7 @@ type AbstractBackend interface {
 	ShouldPostProcess() bool
 
 	// NewEncoder creates an encoder of a TiDB table.
-	NewEncoder(tbl table.Table, options *SessionOptions) Encoder
+	NewEncoder(tbl table.Table, options *SessionOptions, chunk *checkpoints.ChunkCheckpoint) Encoder
```

Comment:
can we make the auto-rand seed as part of the `SessionOptions`? this should reduce a lot of diff.